packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -112,15 +114,17 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
         !areEqual(prevIDs, nextIDs)
       ) {
         this._release();
-        this._localVariables = null;
+        this._localVariables = this._getFragmentVariables();
         this._resolver = createFragmentSpecResolver(
           relay,
           fragments,
           nextProps,
           this._handleFragmentDataUpdate,
         );
         this.setState({relayProp: this._buildRelayProp(relay)});
-      } else if (!this._localVariables) {
+      } else if (true && !this._localVariables) {
+        // question: why did we only call `setProps` if contains hasn't refetched?
+        //           shell we always keep _resolver updated no matter whehter comtains has refetched or not?        
```

Comment:
thanks for explanation. my understanding is that, `_resolver` knows about 
1. dataID 2. query node 3. variables

a change of `dataId` has already been catered for in `areEqual(prevIDs, nextIDs)` ;
query node won't change because they are statically constructed during building;
variables can change as a result of `refetch`

so what is the scenario, where a container hasn't called `refetch`, but needs to haves its `_resolver` updated?